easybuild/easyconfigs/r/R/DMCfun-1.3.0_fix-parallel-detect.patch

```diff
@@ -0,0 +1,37 @@
+# parallal::detectCores() detects physical cores and not available cores. It is ususally problem in queue systems
+# like PBS or Slurm. This patch gives solution for that problem for PBS and with Slurm with PBS compatibility switched
+# on. For other queue systems, a different solutiom might needed.
+# April 13th 2021 by B. Hajgato (UGent)
+diff -ru DMCfun.orig/MD5 DMCfun/MD5
+--- DMCfun.orig/MD5	2021-01-10 18:50:10.000000000 +0100
++++ DMCfun/MD5	2021-04-13 09:31:31.242654188 +0200
+@@ -4,7 +4,7 @@
+ a835c7e49694b2c835c438362b16946f *R/DMCfun.R
+ 034299149a3219bf581d440b19d94fbb *R/RcppExports.R
+ 4e07b13ccaf91457dfc167540129c61f *R/dmcData.R
+-0bce9adf54182001c9eadbca3c6af0f8 *R/dmcFit.R
++1f8c6ded3317c4efee6cdd8538907acc *R/dmcFit.R
+ 655fea3d9c03b02d6f2539ec478ab9b0 *R/dmcSim.R
+ f301185baf6007e9ade686f9964ecf23 *R/methods.R
+ 1b5964cab6608330647f601921f15a5c *R/plot.R
+diff -ru DMCfun.orig/R/dmcFit.R DMCfun/R/dmcFit.R
+--- DMCfun.orig/R/dmcFit.R	2020-12-03 18:53:03.000000000 +0100
++++ DMCfun/R/dmcFit.R	2021-04-13 09:28:47.506520017 +0200
+@@ -159,7 +159,7 @@
+     if (nzchar(chk) && (chk == "true")) {
+       num_cores <- 2L
+     } else {
+-      num_cores <- parallel::detectCores() / 2
++      num_cores <- as.numeric(Sys.getenv("PBS_NP", parallel::detectCores() / 2))
```

Comment:
Why not extend this to 
Sys.getenv("PBS_NP", Sys.getenv("SLURM_CPUS_ON_NODE", parallel::detectCores() / 2))

Reply:
Or possibly the other way around seing that Slurm is more common than PBS these days.

Reply:
I have added `num_cores <- as.numeric(Sys.getenv("SLURM_NTASKS", Sys.getenv("PBS_NP", parallel::detectCores() / 2))`
I think SLURM_NTASKS are closer to the meaning of PBS_NP, which is the number of processors (nodes=x:ppn=y, so PBS_NP=X*Y)

Reply:
Ah, so it is MPI aware?

Reply:
no.

Reply:
Then SLURM_CPUS_ON_NODE is probably more correct.

Reply:
Honestly, I find this hairsplitting, but be it. All in all, if the user incorrectly assumes from an only OpenMP software that it is MPI, then both versions are wrong. You can argue which one is worst, though, but it does not change the fact that both are incorrect. If it is used in a correct way, both versions are just fine. 
Just telling in advance, I will not compare  SLURM_CPUS_ON_NODE  and  SLURM_NTASKS with each other and if they differ, then program will not stop nor write a friendly message that please do not use multiple nodes, because this software is only for one node.